### DISCUSSION TITLE
Text Field - hasMany support

### DISCUSSION THREAD

**Discussion by jmikrut** [payloadcms/payload#4393] [Roadmap]
We will replicate the functionality that was built into the `number` field in order to support `hasMany: true` within `text` fields.

Everything is already done for the `number` field, so this should be a pretty straightforward PR for someone to make! Otherwise, we'll get to this as soon as we can.

**Comment by Cuzart** [comment]
I am going to open a PR after the holidays. I already played around with this and it looks promising but needs some polishing

**Comment by Gokulsck** [comment]
Hey, @jmikrut @Cuzart  I have opened a PR for this. Please check and confirm if it's good and has not missed anything. 

https://github.com/payloadcms/payload/pull/4605/files

**Comment by maxsynnott** [comment]
Just linking [my PR comment](https://github.com/payloadcms/payload/pull/4605#issuecomment-1879146568) to this discussion as it's maybe a better place for it.

Currently the documentation and functionality are out of sync
